codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -495,20 +518,22 @@ private void incrementPendingBytes(int numBytes) {
              * Note: this does not flush the {@link ChannelHandlerContext}.
              */
             int write(int allowedBytes) {
-                int before = payload.size();
-                needFlush |= payload.write(Math.max(0, allowedBytes));
-                int writtenBytes = before - payload.size();
                 try {
+                    pendingWriteQueue.remove();
```

Comment:
@Scottmitch the problem is that a call to `closeLocalSite` cancels every frame in the `pendingWriteQueue`. Now that `closeLocalSite` is called within the `write` method, this would also cancel the stream that was just written (aka the stream with the END_STREAM flag). It's afaik not related to `error`.